ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/LedgerApiServer.scala

```diff
@@ -36,19 +36,20 @@ trait ApiServer {
 
 }
 
-class LedgerApiServer(
+final class LedgerApiServer(
     createApiServices: (Materializer, ExecutionSequencerFactory) => Future[ApiServices],
     desiredPort: Int,
     maxInboundMessageSize: Int,
     address: Option[String],
-    loggerFactory: NamedLoggerFactory,
     sslContext: Option[SslContext] = None,
     interceptors: List[ServerInterceptor] = List.empty,
     metrics: MetricRegistry,
-)(implicit actorSystem: ActorSystem, materializer: Materializer)
+)(implicit actorSystem: ActorSystem, materializer: Materializer, logCtx: LoggingContext)
     extends ResourceOwner[ApiServer] {
+
+  private val logger = ContextualizedLogger.get[this.type]
```

Comment:
Prone to copy-paste bugs. Using `this.type` means if you copy and paste it to a different class, it does the right thing.

Reply:
Since `this.type` has specific semantics and a few (apparently) weird behaviors in edge case, we opted to replace this with a simpler `this.getClass` and a factory method that takes a `Class[_]`.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4097/commits/2e6c24a24c6f76815a2ee04be8aad65264d66bde

Reply:
What are the weird behaviors?